examples/custom_collector.rs

```diff
@@ -102,11 +102,12 @@ struct StatsSegmentCollector {
 impl SegmentCollector for StatsSegmentCollector {
     type Fruit = Option<Stats>;
 
-    fn collect(&mut self, doc: u32, _score: Score) {
+    fn collect(&mut self, doc: u32, _score: Score) -> tantivy::Result<()> {
         let value = self.fast_field_reader.get(doc) as f64;
         self.stats.count += 1;
         self.stats.sum += value;
         self.stats.squared_sum += value * value;
+        Ok(())
```

Comment:
Can you run the bench before and after this change to check it does not impact the compiler perf? 
(Top + Count on "the" term query)